pkg/v1/daemon/image.go

```diff
@@ -85,5 +92,98 @@ func Image(ref name.Reference, options ...Option) (v1.Image, error) {
 		ctx:      o.ctx,
 	}
 
-	return tarball.Image(i.opener(), nil)
+	return &image{
+		ref:    ref,
+		opener: i,
+	}, nil
+}
+
+func (i *image) initialize() error {
+	// Don't re-initialize tarball if already initialized.
+	if i.tarballImage == nil {
+		var err error
+		i.tarballImage, err = tarball.Image(i.opener.opener(), nil)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (i *image) Layers() ([]v1.Layer, error) {
+	if err := i.initialize(); err != nil {
+		return nil, err
+	}
+	return i.tarballImage.Layers()
+}
+
+func (i image) MediaType() (types.MediaType, error) {
+	if err := i.initialize(); err != nil {
+		return "", err
+	}
+	return i.tarballImage.MediaType()
+}
+
+func (i *image) Size() (int64, error) {
+	if err := i.initialize(); err != nil {
+		return 0, err
+	}
+	return i.tarballImage.Size()
+}
+
+func (i *image) ConfigName() (v1.Hash, error) {
+	res, _, err := i.opener.client.ImageInspectWithRaw(i.opener.ctx, i.ref.String())
+	if err != nil {
+		return v1.Hash{}, err
+	}
+	return v1.NewHash(res.ID)
+}
+
+func (i image) ConfigFile() (*v1.ConfigFile, error) {
+	if err := i.initialize(); err != nil {
+		return nil, err
+	}
+	return i.tarballImage.ConfigFile()
+}
+
+func (i image) RawConfigFile() ([]byte, error) {
+	if err := i.initialize(); err != nil {
+		return nil, err
+	}
+	return i.tarballImage.RawConfigFile()
+}
+
+func (i image) Digest() (v1.Hash, error) {
+	if err := i.initialize(); err != nil {
+		return v1.Hash{}, err
+	}
+	return i.tarballImage.Digest()
+}
+
+func (i image) Manifest() (*v1.Manifest, error) {
+	if err := i.initialize(); err != nil {
+		return nil, err
+	}
+	return i.tarballImage.Manifest()
+}
+
+func (i image) RawManifest() ([]byte, error) {
+	if err := i.initialize(); err != nil {
+		return nil, err
+	}
+	return i.tarballImage.RawManifest()
+}
+
+func (i image) LayerByDigest(h v1.Hash) (v1.Layer, error) {
+	if err := i.initialize(); err != nil {
+		return nil, err
+	}
+	return i.tarballImage.LayerByDigest(h)
+}
+
+func (i image) LayerByDiffID(h v1.Hash) (v1.Layer, error) {
+	if err := i.initialize(); err != nil {
+		return nil, err
+	}
+	return i.tarballImage.LayerByDigest(h)
```

Comment:
Should be `LayerByDiffID`

Reply:
Thanks. Fixed https://github.com/google/go-containerregistry/blob/bb5ea5efd13b203c12b6a174e2e9e168cb1032a5/pkg/v1/daemon/image.go